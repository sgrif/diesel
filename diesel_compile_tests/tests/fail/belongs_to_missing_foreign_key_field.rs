#[macro_use]
extern crate diesel;

struct Bar;

#[derive(Associations)]
#[diesel(belongs_to(Bar))]
#[diesel(belongs_to(Bar, foreign_key = bar_id))]
struct Foo {}

#[derive(Associations)]
#[diesel(belongs_to(Bar))]
#[diesel(belongs_to(Bar, foreign_key = bar_id))]
struct Baz {
    #[diesel(column_name = baz_id)]
    bar_id: i32,
}

fn main() {}
