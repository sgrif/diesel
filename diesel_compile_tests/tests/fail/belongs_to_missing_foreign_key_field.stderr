error: No field with column name bar_id
 --> $DIR/belongs_to_missing_foreign_key_field.rs:7:21
  |
7 | #[diesel(belongs_to(Bar))]
  |                     ^^^

error: No field with column name bar_id
  --> $DIR/belongs_to_missing_foreign_key_field.rs:12:21
   |
12 | #[diesel(belongs_to(Bar))]
   |                     ^^^
